module/VuFindSearch/src/VuFindSearch/Backend/Primo/RestConnector.php

```diff
@@ -0,0 +1,705 @@
+     * $this->client and returns the parsed response
+     *
+     * @param string $institution Institution
+     * @param array  $terms       Associative array:
+     *     index       string: primo index to search (default "any")
+     *     lookfor     string: actual search terms
+     * @param array  $params      Associative array of optional arguments:
+     *     phrase      bool:   true if it's a quoted phrase (default false)
+     *     onCampus    bool:   (default true)
+     *     didyoumean  bool:   (default false)
+     *     filterList  array:  (field, value) pairs to filter results (def null)
+     *     pageNumber  string: index of first record (default 1)
+     *     limit       string: number of records to return (default 20)
+     *     sort        string: value to be used by for sorting (default null)
+     *     highlight   bool:   whether to highlight search term matches in records
+     *     highlightStart string: Prefix for a highlighted term
+     *     highlightEnd   string: Suffix for a Highlighted term
+     *     Anything in $params not listed here will be ignored.
+     *
+     * Note: some input parameters accepted by Primo are not implemented here:
+     *  - dym (did you mean)
+     *  - more (get more)
+     *  - lang (specify input language so engine can do lang. recognition)
+     *  - displayField (has to do with highlighting somehow)
+     *
+     * @throws \Exception
+     * @return array             An array of query results
+     *
+     * @link http://www.exlibrisgroup.org/display/PrimoOI/Brief+Search
+     */
+    public function query($institution, $terms, $params = null)
+    {
+        // defaults for params
+        $args = [
+            'phrase' => false,
+            'onCampus' => true,
+            'didYouMean' => false,
+            'filterList' => null,
+            'pcAvailability' => false,
+            'pageNumber' => 1,
+            'limit' => 20,
+            'sort' => null,
+            'highlight' => false,
+            'highlightStart' => '',
+            'highlightEnd' => '',
+        ];
+        if (isset($params)) {
+            $args = array_merge($args, $params);
+        }
+        // Ensure limit is at least 1 since Primo seems to be flaky with 0:
+        $args['limit'] = max(1, $args['limit']);
+
+        return $this->performSearch($terms, $args);
+    }
+
+    /**
+     * Retrieves a document specified by the ID.
+     *
+     * @param string  $recordId  The document to retrieve from the Primo API
+     * @param ?string $inst_code Institution code (optional)
+     * @param bool    $onCampus  Whether the user is on campus
+     *
+     * @throws \Exception
+     * @return array             An array of query results
+     */
+    public function getRecord(string $recordId, $inst_code = null, $onCampus = false)
+    {
+        if ('' === $recordId) {
+            return self::$emptyQueryResponse;
+        }
+        // Query String Parameters
+        $qs   = [];
```

Comment:
Is there a reason for extra spaces here?

Reply:
Legacy. :) Fixed.